articles/quickstart/backend/ruby/01-authentication.md

```diff
@@ -0,0 +1,139 @@
+
+```rb
+# lib/jwt/json_web_token.rb
+
+# frozen_string_literal: true
+require 'net/http'
+require 'uri'
+
+class JsonWebToken
+  def self.verify(token)
+    JWT.decode(token, nil,
+               true, # Verify the signature of this token
+               algorithm: 'RS256',
+               iss: Sinatra::Application.settings.auth0_domain,
+               verify_iss: true,
+               aud: Sinatra::Application.settings.auth0_api_audience,
+               verify_aud: true) do |header|
+      jwks_hash[header['kid']]
+    end
+  end
+
+  def self.jwks_hash
+    jwks_raw = Net::HTTP.get URI("#{Sinatra::Application.settings.auth0_domain}.well-known/jwks.json")
+    jwks_keys = Array(JSON.parse(jwks_raw)['keys'])
+    Hash[
+      jwks_keys
+      .map do |k|
+        [
+          k['kid'],
+          OpenSSL::X509::Certificate.new(
+            Base64.decode64(k['x5c'].first)
+          ).public_key
+        ]
+      end
+    ]
+  end
+end
+```
+
+## Define a Secured Concern
+
+Create an `authenticate!` method to run before each endpoint which looks for the `access_token` in the `Authorization` header of an incoming request. If the token is present, it should be passed to `JsonWebToken.verify`.
+
+```rb
+# lib/server_rs256.rb
+
+def authenticate!
+  # Extract <token> from the 'Bearer <token>' value of the Authorization header
+  supplied_token = String(request.env['HTTP_AUTHORIZATION']).slice(7..-1)
+
+  @auth_payload, @auth_header = JsonWebToken.verify(supplied_token)
+
+rescue JWT::DecodeError => e
+  halt 401, json(error: e.class, message: e.message)
+end
+
+before do
+  authenticate!
+end
+
+get '/restricted_resource' do
+  json( message: 'Access Granted', allowed_scopes: String(@auth_payload['scope']).split(' ') )
+end
+```
+
+## Configuring Scopes
+
+The `JsonWebToken.verify` method above verifies that the `access_token` included in the request is valid; however, it doesn't yet include any mechanism for checking that the token has the sufficient **scope** to access the requested resources.
+
+Scopes provide a way for you to define which resources should be accessible by the user holding a given `access_token`. For example, you might choose to permit `read` access to a `messages` resource if a user has a **manager** access level, or a `write` access to that resource if they are an **administrator**.
+
+To configure scopes in your Auth0 dashboard, navigate to [your API](${manage_url}/#/apis) and choose the **Scopes** tab. In this area you can apply any scopes you wish, including one called `read:messages`, which will be used in this example.
+
+## Protect Individual Endpoints
+
+To look for a particular `scope` provide an array of required scopes and check if the required ones are present in the given `access_token` by being part of the `payload`.
```

Comment:
Let's reword this paragraph to:

To look for a particular `scope` in an `access_token`, provide an array of required scopes and check if they are present in the payload of the token.